cli/command/plugin/create_test.go

```diff
@@ -0,0 +1,150 @@
+		testutil.ErrorContains(t, cmd.Execute(), tc.expectedError)
+	}
+}
+
+func TestCreateErrorOnFileAsContextDir(t *testing.T) {
+	tmpFile, err := ioutil.TempFile("", "file-as-context-dir")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpFile.Name())
+
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpFile.Name()})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "context must be a directory")
+}
+
+func TestCreateErrorOnContextDirWithoutConfig(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpDir})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "config.json: no such file or directory")
+}
+
+func TestCreateErrorOnInvalidConfig(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = setupPluginDir(tmpDir, []byte("invalid-config-contents"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpDir})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "invalid")
+}
+
+func TestCreateErrorFromDaemon(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = setupPluginDir(tmpDir, []byte(`{ "Name": "plugin-foo" }`))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cli := test.NewFakeCli(&fakeClient{
+		pluginCreateFunc: func(createContext io.Reader, createOptions types.PluginCreateOptions) error {
+			return fmt.Errorf("Error creating plugin")
+		},
+	})
+
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpDir})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "Error creating plugin")
+}
+
+func setupPluginDir(dir string, data []byte) error {
+	rootfs := filepath.Join(dir, "rootfs")
+	if err := os.Mkdir(rootfs, 0600); err != nil {
+		return err
+	}
+
+	configFile := filepath.Join(dir, "config.json")
+	return ioutil.WriteFile(configFile, data, 0600)
+}
+
+func TestCreatePlugin(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = setupPluginDir(tmpDir, []byte(`{ "Name": "plugin-foo" }`))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cli := test.NewFakeCli(&fakeClient{
+		pluginCreateFunc: func(createContext io.Reader, createOptions types.PluginCreateOptions) error {
+			return nil
+		},
+	})
+
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpDir})
+	assert.NoError(t, cmd.Execute())
+	assert.Equal(t, "plugin-foo", strings.TrimSpace(cli.OutBuffer().String()))
```

Comment:
I think we should check the stdout without `TrimSpace()`. The trailing newline is important.

There's a few instances of this below as well.

Reply:
Thanks for the review. I have updated the PR accordingly.